contract/models/contract.py

```diff
@@ -517,8 +517,10 @@ def _prepare_recurring_invoices_values(self, date_ref=False):
                 if invoice_line_vals:
                     # Allow extension modules to return an empty dictionary for
                     # nullifying line. We should then cleanup certain values.
-                    del invoice_line_vals["company_id"]
-                    del invoice_line_vals["company_currency_id"]
+                    if invoice_line_vals.get('company_id'):
```

Comment:
@aswani-beo 

I guess we should not check the value for the presence of the key. Otherwise it won't delete e.g. the company_id if it is False.

Reply:
if 'company_id ' in invoice_line_vals:
    del invoice_line_vals["company_id"]